cmd/proxy/console.go

```diff
@@ -0,0 +1,266 @@
+	router.POST("/api/v1/consoles", createConsole)
+}
+
+func findById(id string) *Console {
+	console, ok := consoleMap[id]
+	if !ok {
+		return nil
+	}
+	return &console
+}
+
+func getConsole(c *gin.Context) {
+	id := c.Param("id")
+	console := findById(id)
+	if console == nil {
+		c.JSON(http.StatusNotFound, gin.H{})
+		return
+	}
+	c.JSON(http.StatusOK, console)
+}
+
+func listConsoles(c *gin.Context) {
+	c.JSON(http.StatusOK, consoleMap)
+}
+
+type TerminalSize struct {
+	Rows int `json:"rows"`
+	Cols int `json:"cols"`
+}
+
+type StartRequest struct {
+	Id   string       `json:"id"`
+	Size TerminalSize `json:"size"`
+}
+
+type StopRequest struct {
+	Id string `json:"id"`
+}
+
+func createConsole(c *gin.Context) {
+	network := os.Getenv("NETWORK")
+	id := fmt.Sprint(uuid.New())
+	console := Console{
+		Id:      id,
+		Network: network,
+	}
+	consoleMap[id] = console
+	e := fmt.Sprintf("console-%s", id)
+	logger.Debugf("[console] Register event %s", e)
+	c.JSON(http.StatusOK, console)
+}
+
+func writeInitScript(network string) {
+	f, err := os.Create("init.bash")
+	if err != nil {
+		logger.Errorf("Failed to write init.bash: %s", err)
+		return
+	}
+	defer f.Close()
+
+	f.WriteString(`\
+cat <<EOF
+
+                           .___           __  .__   
+          ___  _____ __  __| _/     _____/  |_|  |  
+          \  \/  /  |  \/ __ |    _/ ___\   __\  |  
+           >    <|  |  / /_/ |    \  \___|  | |  |__
+          /__/\_ \____/\____ |     \___  >__| |____/
+                \/          \/         \/           
+--------------------------------------------------------------
+
+EOF
+
+export NETWORK=` + network + `
+export PS1="$NETWORK > "
+function start() {
+	docker start ${NETWORK}_${1}_1 
+}
+function stop() {
+	docker stop ${NETWORK}_${1}_1
+}
+function restart() {
+	docker restart ${NETWORK}_${1}_1
+}
+function down() {
+	echo "Not implemented yet!"
+}
+function logs() {
+	docker logs --tail=100 ${NETWORK}_${1}_1
+}
+function report() {
+	cat <<EOF
+Please click on https://github.com/ExchangeUnion/xud/issues/\
+new?assignees=kilrau&labels=bug&template=bug-report.md&title=Short%2C+concise+\
+description+of+the+bug, describe your issue, drag and drop the file "${NETWORK}\
+.log" which is located in "{logs_dir}" into your browser window and submit \
+your issue.
+EOF
+}
+function xucli() {
+	docker exec -it ${NETWORK}_xud_1 xucli $@
+}
+function lndbtc-lncli() {
+	docker exec -it ${NETWORK}_lndbtc_1 lncli -n ${NETWORK} -c bitcoin $@
+}
+function lndltc-lncli() {
+	docker exec -it ${NETWORK}_lndltc_1 lncli -n ${NETWORK} -c litecoin $@
+}
+function geth() {
+	docker exec -it ${NETWORK}_geth_1 geth $@
+}
+function bitcoin-ctl() {	
+	if [[ $NETWORK == "testnet" ]]; then
+		docker exec -it ${NETWORK}_bitcoind_1 -testnet -user xu -password xu bitcoind $@
+	else
+		docker exec -it ${NETWORK}_bitcoind_1 -user xu -password xu bitcoind $@
+	fi
+}
+function litecoin-ctl() {
+	if [[ $NETWORK == "testnet" ]]; then
```

Comment:
Should this also include simnet?

Reply:
@erkarl There is no bitcoind or litecoind in simnet. So there should be a kind warning when user try to use bitcoin-ctl or litecoin-ctl inside simnet.